CMakeLists.txt

```diff
@@ -493,6 +606,55 @@ if(NANOARROW_BUILD_TESTS)
     gtest_discover_tests(nanoarrow_ipc_files_test)
     gtest_discover_tests(nanoarrow_ipc_hpp_test)
   endif()
+
+  if(NANOARROW_DEVICE)
+    enable_testing()
```

Comment:
There are a few things here that are duplicative of what happens in the IPC branch directly preceding it, but didn't try to solve that as part of this PR. Figured a copy/paste from the old structure into the new would be easiest to review